src/server/page_server.cpp

```diff
@@ -386,13 +448,77 @@ page_server::follower_connection_handler::follower_connection_handler (cubcomm::
   nullptr,
   nullptr)); // TODO handle abnormal disconnection.
 
+  m_ps.get_follower_responder ().register_connection (m_conn.get ());
+
   m_conn->start ();
 }
 
-void page_server::follower_connection_handler::receive_dummy_request (follower_server_conn_t::sequenced_payload
-    &&a_sp)
+void
+page_server::follower_connection_handler::receive_log_pages_fetch (follower_server_conn_t::sequenced_payload &&a_sp)
+{
+  auto log_serving_func = std::bind (&page_server::follower_connection_handler::serve_log_pages, std::ref (*this),
+				     std::placeholders::_1, std::placeholders::_2);
+  m_ps.get_follower_responder ().async_execute (std::ref (*m_conn), std::move (a_sp), std::move (log_serving_func));
+}
+
+void
+page_server::follower_connection_handler::serve_log_pages (THREAD_ENTRY &, std::string &payload_in_out)
+{
+  // Unpack the message data
+  cubpacking::unpacker payload_unpacker { payload_in_out.c_str (), payload_in_out.size () };
+  LOG_PAGEID start_pageid;
+  int cnt;
+
+  assert (payload_in_out.size () == (sizeof (LOG_PAGEID) + OR_INT_SIZE));
+  payload_unpacker.unpack_bigint (start_pageid);
+  payload_unpacker.unpack_int (cnt);
+
+  const bool perform_logging = prm_get_bool_value (PRM_ID_ER_LOG_READ_LOG_PAGE);
+
+  // Pack the requested log pages
+  int error = NO_ERROR;
+  payload_in_out = { reinterpret_cast<const char *> (&error), sizeof (error) }; // Pack NO_ERROR assuming it'll succeed.
+  log_reader lr { LOG_CS_SAFE_READER };
+
+  for (int i = 0; i < cnt; i++)
+    {
+      const log_lsa fetch_lsa { start_pageid + i, 0 };
+
+      assert (fetch_lsa.pageid != LOGPB_HEADER_PAGE_ID);
+
+      error = lr.set_lsa_and_fetch_page (fetch_lsa);
+
+      if (error != NO_ERROR)
+	{
+	  // All or Nothing. Abandon all appended pages and just set the error.
+	  payload_in_out = { reinterpret_cast<const char *> (&error), sizeof (error) };
+	  break;
+	}
+      else
+	{
+	  payload_in_out.append (reinterpret_cast<const char *> (lr.get_page ()), LOG_PAGESIZE);
+	}
+    }
+
+  if (perform_logging)
+    {
+      _er_log_debug (ARG_FILE_LINE,
+		     "[READ LOG] Sending log pages to the pager server (%s). Page Id from %lld to %lld, Error code: %d\n",
+		     m_conn->get_underlying_channel_id ().c_str (), start_pageid, start_pageid + cnt - 1, error);
+    }
+}
+
+page_server::follower_connection_handler::~follower_connection_handler ()
```

Comment:
Instances of both `follower_connection_handler` and `followee_connection_handler` must be removed as soon as they are done.
Because underneath, they keep resources open: sockets, threads, ....

If there is no follow-up issue that deals with this, please add. If there is, my bad, please excuse. :)

Reply:
They will removed in follow-up issues. :)